sycl/source/detail/error_handling/enqueue_kernel.cpp

```diff
@@ -21,6 +21,35 @@ namespace detail {
 
 namespace enqueue_kernel_launch {
 
+bool L0HandleInvalidWorkGroupSize(const device_impl &DeviceImpl,
```

Comment:
There should be no L0-specifc in this checking, it should be covered by PI API. Can you combine this with `oclHandleInvalidWorkGroupSize` into just `HandleInvalidWorkGroupSize`?

Reply:
Hi Sergey, could you please see this section here https://github.com/intel/llvm/pull/1998/files#diff-5b57d29ae1eb53bd221ca97692974846R48? I think that is valid for L0 and should be added here.

Reply:
Hi @smaslov-intel , If you look at the current function` L0HandleInvalidWorkGroupSize` the code is doing the same thing as a couple parts of `oclHandleInvalidWorkGroupSize`. So agreed they should be combined. I spoke to Rehana offline and we discussed renaming `oclHandleInvalidWorkGroupSize` to the generic `HandleInvalidWorkGroupSize` and just adding conditional statements  `if (PlatformName.find("OpenCL") != std::string::npos)` around all the checks that are openCL specific. Would this be the best approach?

Reply:
Probably, yes. I'd still want to see if any backend specifics can be hidden under PI, so that those specifics are implemented in the relevant plugins not in in SYCL RT.